cmd/process-agent/collector.go

```diff
@@ -334,16 +354,10 @@ func (l *Collector) run(exit chan struct{}) error {
 			case <-heartbeat.C:
 				statsd.Client.Gauge("datadog.process.agent", 1, tags, 1) //nolint:errcheck
 			case <-queueSizeTicker.C:
-				updateQueueBytes(l.processResults.Weight(), l.rtProcessResults.Weight(), l.podResults.Weight())
-				updateQueueSize(l.processResults.Len(), l.rtProcessResults.Len(), l.podResults.Len())
+				updateQueueBytes(l.queueBytesInfo())
+				updateQueueSize(l.queueSizeInfo())
```

Comment:
We are always updating them in the same spot, why not just have a single `updateQueueStats` call performing these updates with a single struct and initialize it here?

```
func updateQueueStats(stats queueStats) {
	infoMutex.Lock()
	defer infoMutex.Unlock()
	infoProcessQueueSize = stats.processQueueSize
....
    infoProcessQueueBytes = stats.processQueueBytes
}
```
Call it with:
```
updateQueueStats(queueStats{
    // initialize everything here, no need for l.queueBytesInfo() or l.queueSizeInfo() helpers, they just obfuscate it.
    processQueueBytes: l.processResults.Weight(),
    // ... the rest of fields
})
```

Reply:
That's a really good point. I've created the `updateQueueStats` function in https://github.com/DataDog/datadog-agent/pull/12460/commits/7908822911f2394e3a90e2549e8a8d94829fad1c.